python/tchannel/thrift/server.py

```diff
@@ -82,23 +87,28 @@ def build_handler(service_module, method_name, handler):
     args_type = getattr(service_module, method_name + '_args')
     result_type = getattr(service_module, method_name + '_result')
 
-    def thrift_handler(request, response, connection):
+    @tornado.gen.coroutine
+    def thrift_handler(request, response, proxy):
         # TODO: Fix arg scheme passing
         # assert request.arg_scheme == 'thrift', (
         #     "Invalid arg scheme %s" % request.arg_scheme
         # )
 
+        body = yield request.get_body()
         args = args_type()
         args.read(
             TBinaryProtocol.TBinaryProtocolAccelerated(
-                TTransport.TMemoryBuffer(request.body)
+                TTransport.TMemoryBuffer(body)
             )
         )
 
         result = result_type()
         try:
+            thrift_args = [
+                getattr(args, spec[2]) for spec in args.thrift_spec[1:]]
```

Comment:
This looks like it will open us up to versioning issues if the spec is expanded.